openssl/pbpal_connect_openssl.c

```diff
@@ -239,7 +239,11 @@ enum pbpal_tls_result pbpal_check_tls(pubnub_t* pb)
     PUBNUB_LOG_TRACE("pb=%p: SSL connected\n", pb);
     socket_set_rcv_timeout(pb->pal.socket, pb->transaction_timeout_ms);
 
+    #if OPENSSL_VERSION_NUMBER < 0x30000000L
     cert = SSL_get_peer_certificate(ssl);
+    #else
+    cert = SSL_get0_peer_certificate(ssl);
```

Comment:
Why `SSL_get0_peer_certificate` used in place of `SSL_get_peer_certificate`?
According to docs:
> SSL_get_peer_certificate() is an alias of SSL_get1_peer_certificate().

`SSL_get0_peer_certificate` doesn't increase the reference count on X509 object, and this doesn't need to be freed to it. In the current implementation, there is release on line **249** (`X509_free(cert)`).

```suggestion
    #if OPENSSL_VERSION_NUMBER < 0x30000000L
    cert = SSL_get_peer_certificate(ssl);
    #else
    cert = SSL_get1_peer_certificate(ssl);
```

Reply:
> SSL_get_peer_certificate() is an alias of SSL_get1_peer_certificate().

I didn't find that info. 
But anyway - calling the `get1` we have to free that function at the end.

Reply:
we free that anyway - my bad